src/process/io/PathSink.cpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#include "PathSink.h"
+
+#include <experimental/filesystem>
+
+using namespace std;
+using namespace std::experimental::filesystem;
+
+PathSink::PathSink(const path &file, bool forceOverwrite) :
```

Comment:
This is a `FileSink` or not? My subjective interpretation a path is more like an URL -- string representing a location, while a File object is more of a representation of a chunk of persisted data in a filesystem that is accessible by a path. Therefore, when accessing data in a file I'd rather think of file than of the more abstract path/filename/url.